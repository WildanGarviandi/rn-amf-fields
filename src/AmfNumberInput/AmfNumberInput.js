import React, { Component } from 'react'
import PropTypes from 'prop-types'
import fieldPropTypes from '../fieldPropTypes'
import { TextField } from 'react-native-material-textfield'

class AmfNumberInput extends Component {

 onChange = val => this.props.onChange(val)

 render() {

   const { label, value, error } = this.props

   const { onChange } = this
		
   return (
     <TextField
       label={label}
       onChangeText={onChange}
       keyboardType="numeric"
       returnKeyType="next"
       value={value || ''}
       error={error}
     />
   )
 }
}

AmfNumberInput.propTypes = {
  ...fieldPropTypes,
  value : PropTypes.string
}

export default AmfNumberInput
