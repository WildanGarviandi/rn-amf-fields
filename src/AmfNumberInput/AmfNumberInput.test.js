import React from 'react'
import AmfNumberInput from './AmfNumberInput'
import renderer from 'react-test-renderer'

test('AmfNumberInput without validation', () => {
  let props = {
    label: 'Label',
    onChangeValue: () => {},
    value: null
  }

  const component = renderer.create(<AmfNumberInput {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

