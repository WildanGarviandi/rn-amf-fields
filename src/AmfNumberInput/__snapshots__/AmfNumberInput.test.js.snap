// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfNumberInput disabled 1`] = `
<View
  onResponderRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={undefined}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 64,
        "overflow": "hidden",
        "paddingBottom": 8,
        "paddingTop": 32,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "borderWidth": 1,
            "bottom": 0,
            "left": -1.5,
            "position": "absolute",
            "right": -1.5,
            "top": -2,
          },
          Object {
            "borderColor": "rgba(0, 0, 0, .38)",
            "borderStyle": "dotted",
          },
        ]
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "position": "absolute",
          "top": 36,
        }
      }
    >
      <Text
        accessible={true}
        active={false}
        allowFontScaling={true}
        animationDuration={225}
        collapsable={undefined}
        disabled={false}
        ellipsizeMode="tail"
        errored={false}
        focused={false}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.38)",
            "fontSize": 16,
          }
        }
      >
        Label
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        autoCapitalize="sentences"
        blurOnSubmit={true}
        disableFullscreenUI={true}
        editable={false}
        error={undefined}
        keyboardType="numeric"
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="next"
        selectionColor="rgb(0, 145, 234)"
        style={
          Array [
            Object {
              "flex": 1,
              "margin": 0,
              "padding": 0,
              "top": 2,
            },
            Object {
              "color": "rgba(0, 0, 0, .38)",
              "fontSize": 16,
              "height": 24,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "flexDirection": "row",
        "height": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(213, 0, 0)",
              "fontSize": 0,
              "opacity": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, .38)",
              "fontSize": 12,
              "opacity": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`AmfNumberInput without validation 1`] = `
<View
  onResponderRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="auto"
  style={undefined}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(0, 0, 0, 0.38)",
        "borderBottomWidth": 0.5,
        "height": 64,
        "paddingBottom": 8,
        "paddingTop": 32,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "position": "absolute",
          "top": 36,
        }
      }
    >
      <Text
        accessible={true}
        active={false}
        allowFontScaling={true}
        animationDuration={225}
        collapsable={undefined}
        disabled={false}
        ellipsizeMode="tail"
        errored={false}
        focused={false}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.38)",
            "fontSize": 16,
          }
        }
      >
        Label
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        autoCapitalize="sentences"
        blurOnSubmit={true}
        disableFullscreenUI={true}
        editable={true}
        error={undefined}
        keyboardType="numeric"
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="next"
        selectionColor="rgb(0, 145, 234)"
        style={
          Array [
            Object {
              "flex": 1,
              "margin": 0,
              "padding": 0,
              "top": 2,
            },
            Object {
              "color": "rgba(0, 0, 0, .87)",
              "fontSize": 16,
              "height": 24,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "flexDirection": "row",
        "height": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(213, 0, 0)",
              "fontSize": 0,
              "opacity": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, .38)",
              "fontSize": 12,
              "opacity": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
